### PR TITLE
Update SimplifyCFG's replacement of phi with its incoming value

### PR DESCRIPTION
**Pull request by meg-gupta**
Previously we replaced phi with its incoming values when all incoming values were the same.
If the phi had itself as incoming value, it wasn't optimized. This PR handles such cases.

### CODE REVIEW

**Comment by meg-gupta** [comment]
Based on top of https://github.com/apple/swift/pull/72780

**Comment by meg-gupta** [comment]
@swift-ci test

**Review by eeckstein — CHANGES_REQUESTED** [review]

File: lib/SILOptimizer/Transforms/SimplifyCFG.cpp [hunk 1]
```diff
@@ -3794,10 +3794,18 @@ static void tryToReplaceArgWithIncomingValue(SILBasicBlock *BB, unsigned i,
...
+      continue;
+    }
+    if (!V) {
+      V = Incoming[Idx];
+      continue;
+    }
+    if (Incoming[Idx] != V) {
       return;
+    }
   }
```
[the remark this answers is not in the document]
Comment (meg-gupta) [reply lib/SILOptimizer/Transforms/SimplifyCFG.cpp:3809]: Thanks! Fixed.

**Review by atrick — APPROVED** [review]
LGTM

**Pull request merged**